pkg/util/pki/parse_test.go

```diff
@@ -174,3 +180,144 @@ func TestDecodePrivateKeyBytes(t *testing.T) {
 		t.Run(test.name, testFn(test))
 	}
 }
+
+type testBundle struct {
+	cert *x509.Certificate
+	pem  []byte
+	pk   crypto.PrivateKey
+}
+
+func mustCreateBundle(t *testing.T, issuer *testBundle, name string) *testBundle {
+	pk, err := GenerateRSAPrivateKey(2048)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	serialNumber, err := rand.Int(rand.Reader, serialNumberLimit)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	template := &x509.Certificate{
+		Version:               3,
+		BasicConstraintsValid: true,
+		SerialNumber:          serialNumber,
+		PublicKeyAlgorithm:    x509.RSA,
+		PublicKey:             pk.Public(),
+		IsCA:                  true,
+		Subject: pkix.Name{
+			CommonName: name,
+		},
+		NotBefore: time.Now(),
+		NotAfter:  time.Now().Add(time.Minute),
+		KeyUsage:  x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
+	}
+
+	var (
+		issuerKey  crypto.PrivateKey
+		issuerCert *x509.Certificate
+	)
+
+	if issuer == nil {
+		// Selfsigned (no issuer)
+		issuerKey = pk
+		issuerCert = template
+	} else {
+		issuerKey = issuer.pk
+		issuerCert = issuer.cert
+	}
+
+	certpem, cert, err := SignCertificate(template, issuerCert, pk.Public(), issuerKey)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	return &testBundle{pem: certpem, cert: cert, pk: pk}
+}
+
+func TestParseCertificateChain(t *testing.T) {
+	root := mustCreateBundle(t, nil, "root")
+	int1 := mustCreateBundle(t, root, "int-1")
+	int2 := mustCreateBundle(t, int1, "int-2")
+	leaf := mustCreateBundle(t, int2, "leaf")
+	random := mustCreateBundle(t, nil, "random")
+
+	joinPEM := func(first []byte, rest ...[]byte) []byte {
+		for _, b := range rest {
+			first = append(first, b...)
+		}
+		return first
+	}
+
+	tests := map[string]struct {
```

Comment:
I wonder if we do also want to test the scenario where the certificate chain contains invalid certs (that cannot be decoded). Although I guess decoding as such is already tested elsewhere.